ngraph/src/ngraph/op/softmax.cpp

```diff
@@ -212,10 +212,8 @@ void op::v1::Softmax::validate_and_infer_types()
                               ") is out of bounds (argument shape: ",
                               input_shape,
                               ").");
-    if (input_shape.is_static())
-        set_output_type(0, get_input_element_type(0), input_shape.to_shape());
-    else
-        set_output_type(0, get_input_element_type(0), PartialShape::dynamic());
+
+    set_output_type(0, get_input_element_type(0), input_shape);
```

Comment:
@GlebKazantaev, could you review this place, please?

Reply:
This looks ok. The output is supposed to match the input's shape and type. With this change inputs like `{?, 3, 224, 224}` will produce an output `{?, 3, 224, 224}` instead of `?`. Only an input with dynamic rank will now produce output `?`.